### PR TITLE
fix(docs): Update `README` Components links to new URL pattern

### PR DESCRIPTION
**Pull request by tulup-conner**
We don't use `HashRouter` anymore, so the links no longer begin with `#`. Also adds anchors to the `Forms` page for each of the components listed (radio, checkbox, and so forth).

## Breaking changes

None.

## Features

None.

## Bug fixes

- [x] Update `README` Components links to new URL pattern

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,7 +18,7 @@
     <a href="https://badge.fury.io/js/flowbite-react">
       <img alt="Latest release" src="https://badge.fury.io/js/flowbite-react.svg">
     </a>
-    <a href="https://flowbite.com/docs/getting-started/license/">
+    <a href="https://flowbite-react.com/getting-started/license/">
```
Comment (bacali95) [inline README.md:21]: This one needs to be reverted

File: README.md [hunk 2]
```diff
@@ -50,7 +50,7 @@ Documentation for `flowbite-react` is not yet finished.
 
 If you want to browse the components, visit [flowbite-react.com](https://flowbite-react.com/).
 
-If you want to learn more about Flowbite, visit [Flowbite docs](https://flowbite.comhttps://flowbite.com/docs/getting-started/introduction/).
+If you want to learn more about Flowbite, visit [Flowbite docs](https://flowbite-react.comhttps://flowbite-react.com/getting-started/introduction/).
```
Comment (bacali95) [inline README.md:53]: The link seems to be broken can you please fi it

File: README.md [hunk 3]
```diff
@@ -373,10 +373,10 @@ Thank you for your interest in helping! Visit our [guide on contributing](https:
 
 If you need the Figma files for the components you can check out our website for more information:
 
-🎨 [Get access to the Figma design files](https://flowbite.com/figma/)
+🎨 [Get access to the Figma design files](https://flowbite-react.com/figma/)
```
Comment (bacali95) [inline README.md:376]: This one should be reverted

File: README.md [hunk 4]
```diff
@@ -373,10 +373,10 @@ Thank you for your interest in helping! Visit our [guide on contributing](https:
...
 
 ## Copyright and license
 
 The Flowbite name and logos are trademarks of Crafty Dwarf Inc.
 
-📝 [Read about the licensing terms](https://flowbite.com/docs/getting-started/license/)
+📝 [Read about the licensing terms](https://flowbite-react.com/getting-started/license/)
```
Comment (bacali95) [inline README.md:382]: This one as well

**Review by bacali95 — CHANGES_REQUESTED** [review]

**Comment by tulup-conner** [comment]
Thank you! I made those changes and also:

- [x] Changed `Sidebar` to not be under construction
- [x] Reverted some other links that were broken (Typography, components that are not yet converted to `flowbite-react` yet)

**Review by bacali95 — APPROVED** [review]

**Comment by bacali95** [comment]
> @bacali95 
> 
> Thank you! I made those changes and also:
> 
> - [x] Changed `Sidebar` to not be under construction
> - [x] Reverted some other links that were broken (Typography, components that are not yet converted to `flowbite-react` yet)

Great thanks, let's merge this one

**Pull request merged**